webapp/channels/src/root.html

```diff
@@ -60,23 +63,123 @@
+                    </mask>
+                    <mask id="LoadingAnimation__spinner-right-half-mask">
+                        <rect x="52" y="0" width="52" height="104" fill="white"/>
+                        <circle class="LoadingAnimation__spinner-mask" r="20" cx="52" cy="52" fill="black"/>
+                    </mask>
+                </defs>
+                <g class="LoadingAnimation__spinner-container">
+                    <g class="LoadingAnimation__spinner">
+                        <circle r="25" cx="52" cy="52" fill="currentColor" mask="url(#LoadingAnimation__spinner-left-half-mask)"/>
+                        <circle r="25" cx="52" cy="52" fill="url(#LoadingAnimation__spinner-gradient)" mask="url(#LoadingAnimation__spinner-right-half-mask)"/>
+                    </g>
+                </g>
+                <g class="LoadingAnimation__compass">
+                    <g class="LoadingAnimation__compass-base-container" mask="url(#LoadingAnimation__base-wipe-mask)">
+                        <circle class="LoadingAnimation__compass-base" r="52" cx="52" cy="52" fill="currentColor" mask="url(#LoadingAnimation__base-mask)"/>
+                    </g>
+                    <g class="LoadingAnimation__compass-needle-container">
+                        <g class="LoadingAnimation__compass-needle">
+                            <g transform="translate(54,46)">
+                                <g transform="translate(-15,-42)">
+                                    <path
+                                        d="M29.9539 1.4977C29.9539 0.670968 29.2827 0 28.4562 0C27.9597 0 27.5192 0.242028 27.2468 0.614415C27.216 0.656555 27.1873 0.700359 27.1609 0.745666L3.66519 32.1191C1.38202 34.7479 0 38.1803 0 41.9355C0 50.207 6.70541 56.9124 14.977 56.9124C23.2485 56.9124 29.9539 50.207 29.9539 41.9355L29.9539 41.9013V1.50252C29.9539 1.50091 29.9539 1.49931 29.9539 1.4977Z"
+                                        fill="currentColor"
+                                    />
+                                </g>
+                            </g>
+                        </g>
+                    </g>
+                </g>
+            </svg>
+        </div>
+    </div>
     <div id='root'>
         <div class='error-screen'>
             <h2>Cannot connect to Mattermost</h2>
             <hr/>
             <p>We're having trouble connecting to Mattermost. If refreshing this page (Ctrl+R or Command+R) does not work, please verify that your computer is connected to the internet.</p>
             <br/>
         </div>
-        <div
-            class='loading-screen'
-            style='position: relative'
-        >
-            <div class='loading__content'>
-                <div class='round round-1'></div>
-                <div class='round round-2'></div>
-                <div class='round round-3'></div>
-            </div>
-        </div>
     </div>
     <div id='root-portal'></div>
     <noscript>
```

Comment:
Is this still visible when JS is disabled or does it get obscured by the loading indicator at all?

Reply:
Yes handled this case as well